XS✔ ◾ Added Drawbacks to GTM rule (#6787)
SSWConsulting/SSW.Rules.Content: rules/use-google-tag-manager/rule.md

![Figure: GA4 is collecting data from the website](tracking-ga4.png)

Learn more on [Google Tag Manager official site](https://marketingplatform.google.com/about/tag-manager/benefits/).

### Drawbacks
While Google Tag Manager is great for non-tech people to easily embed tracking code, you must be careful to not introduce functionality to the website that may interfere with existing systems. 

Some things to consider when adding GTM scripts to a container:

> * **Performance** - some scripts may have a large performance impact, and result in the slowing of the site. 

Reviewer: any examples to mention here?

> * **Performance** - some scripts may have a large performance impact, and result in the slowing of the site. 

Reviewer 2: ```suggestion
* **Performance** - some scripts may have a large performance impact, and result in the slowing of the site. The more scripts you add, the slower the site becomes
```
checked by @camillars @Harry-Ross @AttackOnMorty

> * **Unexpected side effects** - scripts may interfere with existing systems on the site

Reviewer 3: This point is kind of vague. Could you elaborate more?

> * **Unexpected side effects** - scripts may interfere with existing systems on the site

Reviewer 2: ```suggestion
```
checked by @camillars @Harry-Ross @AttackOnMorty

> * **Duplication of code** - some functionality may already be implemented by developers within the site, which may cause a duplication of logic.

Reviewer: how can devs stop this from happening?

> * **Duplication of code** - some functionality may already be implemented by developers within the site, which may cause a duplication of logic.

Reviewer 2: ```suggestion
```

> * **Disconnected functionality** - some functionality may require developer modifications to ensure that it works correctly, and these customizations may enhance user experience.

Reviewer: again, any examples? this is good to mention but at SSW we do have access to devs, so maybe we need an example.

> * **Disconnected functionality** - some functionality may require developer modifications to ensure that it works correctly, and these customizations may enhance user experience.

Reviewer 2: ```suggestion
* **Disconnected functionality** - some functionality may require developer modifications in the code that are not possible in GTM to ensure that it works correctly. (e.g. Application Insights)
```
